### PR TITLE
Bug 1614039 - Re-write pin jobs tests

### PR DESCRIPTION
**Pull request by camd**
Dang, this took longer than I expected.  But finally got it working.

I opted not to upgrade `react-hotkeys` at this time as it breaks some hot keys (like `?`).  So we can address that in another PR.

### CODE REVIEW

File: ui/job-view/App.jsx [hunk 1]
```diff
@@ -378,7 +378,6 @@ class App extends React.Component {
                   <DetailsPanel
                     resizedHeight={detailsHeight}
                     currentRepo={currentRepo}
-                    repoName={repoName}
```
Comment (camd) [inline ui/job-view/App.jsx]: This was not necessary and just noise...

File: ui/shared/HelpMenu.jsx [hunk 2]
```diff
@@ -80,6 +80,7 @@ const HelpMenu = () => (
           target="_blank"
           rel="noopener noreferrer"
           href={item.href}
+          key={item.text}
```
Comment (camd) [inline ui/shared/HelpMenu.jsx]: This was throwing those "need a uniquey key errors" messages.  So I fixed it here.

File: tests/ui/job-view/details/PinBoard_test.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            />
+          </div>
+        </KeyboardShortcuts>
+      </Provider>
+    </div>
+  );
+
+  test('pin selected job with button', async () => {
+    const { store } = configureStore();
+    store.dispatch(setPushes(pushFixture.results, {}));
```
Comment (armenzg) [inline tests/ui/job-view/details/PinBoard_test.jsx]: Does it make sense to move these two lines to `beforeEach`? (and the same lines on the other tests)

File: tests/ui/job-view/details/PinBoard_test.jsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    const { getByTitle, getByTestId } = render(testDetailsPanel(store));
+    store.dispatch(setSelectedJob(jobList.data[1], true));
+
+    const hotkeys = await waitForElement(() => getByTestId('hot-keys-id'));
+    const keyDownEvent = new Event('keydown');
+    keyDownEvent.keyCode = 32;
+    keyDownEvent.key = 'Space';
+
+    hotkeys.focus();
+    hotkeys.dispatchEvent(keyDownEvent);
```
Comment (armenzg) [inline tests/ui/job-view/details/PinBoard_test.jsx]: Very cool 😄

File: tests/ui/job-view/details/PinBoard_test.jsx (hunk 3 above)
Comment (camd) [reply tests/ui/job-view/details/PinBoard_test.jsx]: Good call!  Done...

**Pull request merged**